src/Shared/FileUtilities.cs

```diff
@@ -824,32 +827,30 @@ internal static FileInfo GetFileInfoNoThrow(string filePath)
         /// </summary>
         /// <param name="fullPath">Full path to the directory in the filesystem</param>
         /// <returns></returns>
-        internal static bool DirectoryExistsNoThrow(string fullPath)
+        internal static bool DirectoryExistsNoThrow(string fullPath
+#if !CLR2COMPATIBILITY
```

Comment:
Would it streamline things to make a CLR2COMPATIBILITY FileSystem that was as basic as possible? We don't care about perf _at all_ in that scenario just about code cleanliness. But if it's more trouble than it's worth this is fine.